### ISSUE TITLE
Rule Change: make the `func-style` rule schema strict

### ISSUE THREAD

**Issue by Tanujkanti4441** [eslint/eslint#20689]
### What rule do you want to change?

`func-style`

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

Other

### Example code

```js
/*eslint func-style: ["error", "expression", { "allowArrowFunctions": true }]*/
```

### What does the rule currently do for this code?

It accepts this config even `"allowArrowFunctions": true` is only allowed with `declaration` option and has no effect with `expression`.

### What will the rule do after it's changed?

It will not accept the config that has no effect on the linting.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

PR https://github.com/eslint/eslint/pull/20687 was raised as user used such config.

**Comment by mdjermanovic** [comment]
Makes sense to me, but since this kind of changes are considered breaking per our policies, I'm adding this to v11 board so we can discuss it for the next major release.
